google-cloud-spanner/src/test/java/com/google/cloud/spanner/AsyncTransactionManagerTest.java

```diff
@@ -983,17 +984,13 @@ public void asyncTransactionManagerWaitsUntilAsyncBatchUpdateHasFinished() throw
         }
       }
     }
+    ImmutableList<Class<? extends GeneratedMessageV3>> expectedRequests =
+        ImmutableList.of(
+            BatchCreateSessionsRequest.class, ExecuteBatchDmlRequest.class, CommitRequest.class);
     if (isMultiplexedSessionsEnabled()) {
-      assertThat(mockSpanner.getRequestTypes())
-          .containsExactly(
-              CreateSessionRequest.class,
-              BatchCreateSessionsRequest.class,
-              ExecuteBatchDmlRequest.class,
-              CommitRequest.class);
+      assertThat(mockSpanner.getRequestTypes()).containsAtLeastElementsIn(expectedRequests);
```

Comment:
AsyncTransactionManager does not support multiplexed sessions yet. Just curious to understand at what condition this flakiness is occurring?

Reply:
Although read/write transactions do not support multiplexed sessions, we do pre-emptively create a multiplexed session asynchronously when a client is created with multiplexed sessions enabled. We do this to try to avoid that the first query that can use multiplexed sessions has to wait for it. That again means that these tests might or might not see a multiplexed session being created (random, because the request is executed async). As the multiplexed session is not used here, we just choose to ignore it in the test, as it is irrelevant.